File: TuyenTruyen/BE/Project.Net8/Models/Major/BaiThiModel.cs
using BaiThi.Models.Major;
using DTC.DefaultRepository.Constants;
using DTC.DefaultRepository.Models.Base;
using DTC.T;
using MongoDB.Bson.Serialization.Attributes;
using Project.Net8.Models.Core;
using CoreModel = DTC.DefaultRepository.Models.Core.CoreModel;

namespace Project.Net8.Models.Major;

public class BaiThiModel: Audit, TEntity<string>
{

    // Tiêu đề là name á nghen
    public string MoTa {  get; set; }
    public DateTime? NgayBatDauTuyenTruyen { get; set; }

    [BsonIgnore]
    public string NgayBatDauTuyenTruyenShow// lấy ra này nè nghen 
    {
        get { return NgayBatDauTuyenTruyen.HasValue ? NgayBatDauTuyenTruyen.Value.ToLocalTime().ToString(FormatTime.FORMAT_DATE_CORE) : ""; }
    }
    public DateTime? NgayKetThuc { get; set; }

    [BsonIgnore]
    public string NgayKetThucShow// lấy ra này nè nghen 
    {
        get { return NgayKetThuc.HasValue ? NgayKetThuc.Value.ToLocalTime().ToString(FormatTime.FORMAT_DATE_CORE) : ""; }
    }
    public string TenDonVi {  get; set; }

    public bool IsDuyet { get; set; } = false;// trường này để duyệt nè

    public List<LichSuTuyenTruyenModel> LichSu { get; set; } = new List<LichSuTuyenTruyenModel>();
}

File: TuyenTruyen/FE/src/models/baiThiModel.js
import moment from "moment";
const toJson = (item) => {
  return {
    id: item.id,
    name: item.name,
    tenDonVi: item.tenDonVi,
    moTa: item.moTa,
    ngayBatDauTuyenTruyen: item.ngayBatDauTuyenTruyen,
    ngayKetThuc: item.ngayKetThuc,
    isDuyet: item.isDuyet,
    lichSu: item.lichSu,
  };
};

const fromJson = (item) => {
  return {
    id: item.id,
    name: item.name,
    tenDonVi: item.tenDonVi,
    moTa: item.moTa,
    ngayBatDauTuyenTruyen: item.ngayBatDauTuyenTruyen,
    ngayKetThuc: item.ngayKetThuc,
    isDuyet: item.isDuyet,
    lichSu: item.lichSu,
  };
};

const baseJson = () => {
  return {
    id: null,
    name: null,
    tenDonVi: null,
    moTa: null,
    ngayBatDauTuyenTruyen: null,
    ngayKetThuc: null,
    isDuyet: false,
    lichSu: null,
  };
};

const toListModel = (items) => {
  if (items.length > 0) {
    let data = [];
    items.map((value, index) => {
      data.push(fromJson(value));
    });
    return data ?? [];
  }
  return [];
};
export const baiThiModel = {
  toJson,
  fromJson,
  baseJson,
  toListModel,
};
